components/dashboard/src/admin/UserDetail.tsx

```diff
@@ -290,8 +360,92 @@ export default function UserDetail(p: { user: User }) {
                         {renderUserBillingProperties()}
                     </div>
                 </div>
+
                 <WorkspaceSearch user={user} />
             </PageWithAdminSubMenu>
+            <Modal
+                visible={editSpendingLimit}
+                onClose={() => setEditSpendingLimit(false)}
+                title="Update Usage Limit"
+                onEnter={() => false}
+                buttons={[
+                    <button
+                        className="secondary"
+                        disabled={usageLimit === costCenter?.spendingLimit}
+                        onClick={async () => {
+                            if (usageLimit !== undefined) {
+                                await getGitpodService().server.adminSetUsageLimit(
+                                    AttributionId.render(AttributionId.create(user)),
+                                    usageLimit || 0,
+                                );
+                                setUsageLimit(undefined);
+                                initialize();
+                                setEditSpendingLimit(false);
+                            }
+                        }}
+                    >
+                        Save
+                    </button>,
+                ]}
+            >
+                <p>Update the spending limit for this account.</p>
+                <div className="flex flex-col">
+                    <input
+                        type="number"
+                        min={usage?.creditsUsed}
+                        max={500000}
+                        title="Update Usage Limit"
+                        value={usageLimit}
+                        onChange={(event) => setUsageLimit(Number.parseInt(event.target.value))}
+                    />
+                </div>
+            </Modal>
+            <Modal
+                onEnter={() => false}
+                visible={editAddCreditNote}
+                onClose={() => setEditAddCreditNote(false)}
+                title="Add Credits"
+                buttons={[
+                    <button
+                        className="secondary"
+                        disabled={creditNote.credits === 0 || !creditNote.note}
+                        onClick={async () => {
+                            if (creditNote.credits !== 0 && !!creditNote.note) {
+                                await getGitpodService().server.adminAddUsageCreditNote(
+                                    AttributionId.render(AttributionId.create(user)),
+                                    creditNote.credits,
+                                    creditNote.note,
+                                );
+                                setEditAddCreditNote(false);
+                                setCreditNote({ credits: 0 });
+                                initialize();
+                            }
+                        }}
+                    >
+                        Save
```

Comment:
**issue**: Probably not the best line to comment about this, but we could probably surface these errors we return in console when entering invalid values. Possibly good for a follow up issue. 🏓 

<img width="1440" alt="Screenshot 2022-11-21 at 2 40 40 PM (2)" src="https://user-images.githubusercontent.com/120486/203057913-8f54765b-8c52-4021-b488-d60cf60c4fcd.png">

Reply:
oh, you should not get that error 👀